`[stream] xclaim and xpending, all options` test seems to be flaky

https://github.com/denodrivers/redis/runs/1523779037
```
failures:

[stream] xclaim and xpending, all options
Error: -ERR The ID specified in XADD is equal or smaller than the target stream top item

    at tryParseErrorReply (io.ts:194:11)
    at readReply (io.ts:104:7)
    at async RedisImpl.execBulkReply (redis.ts:99:24)
    at async Promise.all (index 0)
    at async stream_test.ts:473:5
    at async withConsumerGroup (stream_test.ts:39:3)
    at async Object.func (stream_test.ts:464:3)
    at async test_util.ts:35:9
    at async asyncOpSanitizer (deno:cli/rt/40_testing.js:36:7)
    at async Object.resourceSanitizer [as fn] (deno:cli/rt/40_testing.js:70:7)

[stream] xinfo
AssertionError: Test case is leaking async ops.
Before:
  - dispatched: 1381
  - completed: 1380
After:
  - dispatched: 1412
  - completed: 1412

Make sure to await all promises returned from Deno APIs before
finishing test case.
    at assert (deno:cli/rt/06_util.js:33:13)
    at asyncOpSanitizer (deno:cli/rt/40_testing.js:46:7)
    at async Object.resourceSanitizer [as fn] (deno:cli/rt/40_testing.js:70:7)
    at async TestRunner.[Symbol.asyncIterator] (deno:cli/rt/40_testing.js:245:13)
    at async Object.runTests (deno:cli/rt/40_testing.js:322:22)
    at async $deno$test.ts:3:1

failures:

	[stream] xclaim and xpending, all options
	[stream] xinfo
```

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -19,10 +19,9 @@ suite.afterAll(() => {
   client.close();
 });
 
+const rnum = () => Math.floor(Math.random() * 1000);
 const randomStream = () =>
-  `test-deno-${(new Date().getTime())}-${Math.floor(Math.random() * 1000)}${
-    Math.floor(Math.random() * 1000)
-  }${Math.floor(Math.random() * 1000)}`;
+  `test-deno-${(new Date().getTime())}-${rnum()}${rnum()}${rnum()}`;
 
 const cleanupStream = async (client: Redis, ...keys: string[]) => {
   await Promise.all(keys.map((key) => client.xtrim(key, { elements: 0 })));

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -20,7 +20,7 @@ suite.afterAll(() => {
 });
 
 const randomStream = () =>
-  `test-deno-${Math.floor(Math.random() * 1000)}${
+  `test-deno-${(new Date().getTime())}-${Math.floor(Math.random() * 1000)}${
     Math.floor(Math.random() * 1000)
   }${Math.floor(Math.random() * 1000)}`;
 

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -19,7 +19,7 @@ suite.afterAll(() => {
   client.close();
 });
 
-const randomStream = () => `test-deno-${Math.floor(Math.random() * 1000)}`;
+const randomStream = () => `test-deno-${Math.floor(Math.random() * 1000)}${Math.floor(Math.random() * 1000)}${Math.floor(Math.random() * 1000)}`;
 
 const cleanupStream = async (client: Redis, ...keys: string[]) => {
   await Promise.all(keys.map((key) => client.xtrim(key, { elements: 0 })));

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -484,7 +484,7 @@ suite.test("xclaim and xpending, all options", async () => {
 
     const firstPending = await client.xpending(key, group);
 
-    await delay(5);
+    await delay(50);
 
     assertEquals(firstPending.count, 2);
     assertNotEquals(firstPending.startId, firstPending.endId);
@@ -531,7 +531,7 @@ suite.test("xclaim and xpending, all options", async () => {
     });
 
     // take a short nap and increase the lastDeliveredMs
-    await delay(5);
+    await delay(50);
 
     // try another form of xpending: counts for all consumers (we have only one)
     const secondPending = await client.xpending_count(key, group, {
@@ -588,7 +588,7 @@ suite.test("xclaim and xpending, all options", async () => {
       consumer: "weird-interloper",
     });
 
-    await delay(5);
+    await delay(50);
 
     // try another form of xpending: counts
     // efficiently filtered down to a single consumer.
@@ -716,7 +716,7 @@ suite.test("xinfo", async () => {
     await client.xreadgroup([[key, ">"]], { group, consumer: "newbie" });
 
     // Increase the idle time by falling asleep
-    await delay(2);
+    await delay(10);
     const consumerInfos = await client.xinfo_consumers(key, group);
     assertEquals(consumerInfos.length, 2);
     const newConsumer = consumerInfos.find((c) => c.name === "newbie");

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -533,7 +533,7 @@ suite.test("xclaim and xpending, all options", async () => {
     });
 
     // take a short nap and increase the lastDeliveredMs
-    await delay(50);
+    await delay(5);
 
     // try another form of xpending: counts for all consumers (we have only one)
     const secondPending = await client.xpending_count(key, group, {
@@ -590,7 +590,7 @@ suite.test("xclaim and xpending, all options", async () => {
       consumer: "weird-interloper",
     });
 
-    await delay(50);
+    await delay(5);
 
     // try another form of xpending: counts
     // efficiently filtered down to a single consumer.
@@ -718,7 +718,7 @@ suite.test("xinfo", async () => {
     await client.xreadgroup([[key, ">"]], { group, consumer: "newbie" });
 
     // Increase the idle time by falling asleep
-    await delay(10);
+    await delay(2);
     const consumerInfos = await client.xinfo_consumers(key, group);
     assertEquals(consumerInfos.length, 2);
     const newConsumer = consumerInfos.find((c) => c.name === "newbie");

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -19,7 +19,10 @@ suite.afterAll(() => {
   client.close();
 });
 
-const randomStream = () => `test-deno-${Math.floor(Math.random() * 1000)}${Math.floor(Math.random() * 1000)}${Math.floor(Math.random() * 1000)}`;
+const randomStream = () =>
+  `test-deno-${Math.floor(Math.random() * 1000)}${
+    Math.floor(Math.random() * 1000)
+  }${Math.floor(Math.random() * 1000)}`;
 
 const cleanupStream = async (client: Redis, ...keys: string[]) => {
   await Promise.all(keys.map((key) => client.xtrim(key, { elements: 0 })));

--- a/tests/stream_test.ts
+++ b/tests/stream_test.ts
@@ -486,7 +486,7 @@ suite.test("xclaim and xpending, all options", async () => {
 
     const firstPending = await client.xpending(key, group);
 
-    await delay(50);
+    await delay(5);
 
     assertEquals(firstPending.count, 2);
     assertNotEquals(firstPending.startId, firstPending.endId);

